app/javascripts/components/playSlot/actions.jsx

```diff
@@ -162,30 +162,35 @@ export function requestToPlayGame(playInfo, stopSpinFunc) {
     dispatch({
       type: ACTION_TYPES.START_TO_PLAY_GAME,
     });
-    try {
-      const transaction = await Web3Service.playSlotMachine(playInfo);
-      const reward = await Web3Service.getSlotResult(playInfo.slotMachineContract);
-      const betMoney = playInfo.lineNum * Web3Service.makeWeiFromEther(playInfo.betSize);
-      const ethReward = Web3Service.makeEthFromWei(reward.args.reward);
-      const diffMoney = reward.args.reward - betMoney;
-      stopSpinFunc(ethReward);
-      dispatch({
-        type: ACTION_TYPES.SUCCEEDED_TO_PLAY_GAME,
-        payload: {
-          transaction,
-          diffMoney,
-        },
-      });
-    } catch (err) {
-      Toast.notie.alert({
-        type: 'error',
-        text: 'There was an error for playing a slot machine',
-        stay: true,
+    const rewardPromise = Web3Service.getSlotResult(playInfo.slotMachineContract);
+    const playPromise = Web3Service.playSlotMachine(playInfo);
+    await Promise.all([rewardPromise, playPromise])
```

Comment:
No, play contract can be decided with reward event. So those promises have to be synchronously executed.